rest/session_api.go

```diff
@@ -157,10 +158,39 @@ func (h *handler) makeSessionFromEmail(email string, createUserIfNeeded bool) er
 			return err
 		}
 	}
+
 	return h.makeSession(user)
 
 }
 
+// makeSessionFromName attempts to find the user by username.
+// If not found, and createUserIfNeeded=true, creates a new user based on username.
+func (h *handler) makeSessionFromName(username string, createUserIfNeeded bool) error {
```

Comment:
I don't think we need a parallel creation option - I think it's sufficient to modify MakeSessionFromUserAndEmail to allow an empty email when in the create processing.  It looks like the only other caller is for google auth, and we don't need to enforce email there either.

Reply:
OK, makes sense if we want that behaviour for the Google auth flow too. I've updated the PR with those changes.